Explorer/Assets/DCL/Chat/ChatCommandsHandler.cs

```diff
@@ -0,0 +1,97 @@
+﻿using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using ECS.SceneLifeCycle.Realm;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading;
+using UnityEngine;
+
+namespace DCL.Chat
+{
+    internal class ChatCommandsHandler
+    {
+        private const char CHAT_COMMAND_CHAR = '/';
+
+        private const string GOTO_KEY = "goto";
+        private const string WORLD_KEY = "world";
+        private const string GENESIS_KEY = "genesis";
+        private const string RANDOM_KEY = "random";
+
+        private static readonly Regex CHANGE_REALM_REGEX = new ("^/(" + WORLD_KEY + "|" + GOTO_KEY + @")\s+(\S+\.dcl\.eth|" + GENESIS_KEY + ")$", RegexOptions.Compiled);
+        private static readonly Regex TELEPORT_REGEX = new ("^/" + GOTO_KEY + @"\s+(?:(-?\d+),(-?\d+)|" + RANDOM_KEY + ")$", RegexOptions.Compiled);
+
+        private readonly IRealmNavigator realmNavigator;
+
+        private readonly URLDomain worldDomain = URLDomain.FromString(IRealmNavigator.WORLDS_DOMAIN);
+
+        private readonly Dictionary<string, URLAddress> worldAddressesCaches = new ();
+
+        public ChatCommandsHandler(IRealmNavigator realmNavigator)
+        {
+            this.realmNavigator = realmNavigator;
+        }
+
+        public bool TryGetChatCommand(in string message, ref UniTask<string> command)
+        {
+            if (!message.StartsWith(CHAT_COMMAND_CHAR)) return false;
+
+            if (CHANGE_REALM_REGEX.IsMatch(message))
+            {
+                command = ChangeRealmCommandAsync(message);
+                return true;
+            }
+
+            if (TELEPORT_REGEX.IsMatch(message))
+            {
+                command = TeleportToCommandAsync(message);
+                return true;
+            }
+
+            return false;
+        }
+
+        private async UniTask<string> TeleportToCommandAsync(string message)
+        {
+            Match match = TELEPORT_REGEX.Match(message);
+
+            int x, y;
+            if (match.Groups[1].Success && match.Groups[2].Success)
+            {
+                x = int.Parse(match.Groups[1].Value);
+                y = int.Parse(match.Groups[2].Value);
+            }
+            else
+            {
+                x = Random.Range(-150, 150);
+                y = Random.Range(-150, 150);
+            }
+
+            await realmNavigator.TeleportToParcelAsync(new Vector2Int(x, y), CancellationToken.None);
+            return $"🟢 You teleported to {x},{y} in Genesis City";
+        }
+
+        private async UniTask<string> ChangeRealmCommandAsync(string message)
+        {
+            Match match = CHANGE_REALM_REGEX.Match(message);
+            string worldName = match.Groups[2].Value;
+            string realmUrl = worldName == GENESIS_KEY ? IRealmNavigator.GENESIS_URL : GetWorldAddress(worldName);
+
+            bool isSuccess = await realmNavigator.TryChangeRealmAsync(realmUrl, CancellationToken.None);
```

Comment:
CancellationToken should be propagated from the outer-most consumer.
You should be handling cancellation carefully as otherwise it is possible to launch multiple realm change flows at the same time and it won't go well.

Reply:
this is a possible good edge-case to add to our test-cases, @anicalbano. For example, it was theoretically possible to run `/goto olavra.dcl.eth` and then fast type `/goto 5,5`, but you need to be faster than a rabbit to achieve it 😅